pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -59,66 +60,162 @@ func (dv *openShiftClusterDynamicValidator) Dynamic(ctx context.Context) error {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to validate the first party principal",
+		)
 	}
 
 	err = fpDynamic.ValidateVnet(ctx, dv.oc.Location, subnets, dv.oc.Properties.NetworkProfile.PodCIDR, dv.oc.Properties.NetworkProfile.ServiceCIDR)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidLinkedVNet,
+			"", "Invalid virtual network specified",
+		)
 	}
 
 	err = fpDynamic.ValidateDiskEncryptionSets(ctx, dv.oc)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Invalid disk encryption sets specified",
+		)
 	}
 
 	spp := dv.oc.Properties.ServicePrincipalProfile
 	token, err := aad.GetToken(ctx, dv.log, spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID, dv.env.Environment().ActiveDirectoryEndpoint, dv.env.Environment().ResourceManagerEndpoint)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to retrieve an AAD token for the service principal",
+		)
 	}
 
 	spAuthorizer := refreshable.NewAuthorizer(token)
 
 	spDynamic, err := dynamic.NewValidator(dv.log, dv.env, dv.env.Environment(), dv.subscriptionDoc.ID, spAuthorizer, dynamic.AuthorizerClusterServicePrincipal)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to validate the service principal token",
+		)
 	}
 
 	// SP validation
 	err = spDynamic.ValidateServicePrincipal(ctx, spp.ClientID, string(spp.ClientSecret), dv.subscriptionDoc.Subscription.Properties.TenantID)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidParameter,
+			"", "Failed to validate the cluster service principal",
+		)
 	}
 
 	err = spDynamic.ValidateVnet(ctx, dv.oc.Location, subnets, dv.oc.Properties.NetworkProfile.PodCIDR, dv.oc.Properties.NetworkProfile.ServiceCIDR)
 	if err != nil {
-		return err
+		if _, isCloudError := err.(*api.CloudError); isCloudError {
+			return err
+		}
+		dv.log.Error(err)
+		return api.NewCloudError(
+			http.StatusBadRequest,
+			api.CloudErrorCodeInvalidLinkedVNet,
+			"", "Invalid cluster virtual network CIDRs",
```

Comment:
`spDynamic.ValidateVnet` can return errors not related to CIDRs. It is the same validation as `fpDynamic.ValidateVnet`, but with different service principal.